### PR TITLE
fix: running check error when podman is default in wsl

### LINKED ISSUE #17227: [Bug]: Does not work if the machine is set to default in wsl.
### Issue Description

When executing the 'podman machine start' command, the following message appears.

``` bat
> podman machine start

API forwarding for Docker API clients is not available due to the following startup failures.
        could not start api proxy since expected pipe is not available: podman-machine-default
```

While looking for the cause, I found out that the above error occurred when podman's WSL distro was set to the default.
The cause will be described in **Additional information** below.

### Steps to reproduce the issue

Steps to reproduce the issue
1. wsl --set-default podman-machine-default
2. podman machine start

### Describe the results you received

```could not start api proxy since expected pipe is not available: podman-machine-default``` messsage appear

### Describe the results you expected

Normal execution of podman

### podman info output

```shell
host:
  arch: amd64
  buildahVersion: 1.28.0
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.5-1.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.5, commit: '
  cpuUtilization:
    idlePercent: 99.88
    systemPercent: 0.07
    userPercent: 0.04
  cpus: 8
  distribution:
    distribution: fedora
    variant: container
    version: "36"
  eventLogger: journald
  hostname: DESKTOP-3H6L2DB
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.15.79.1-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 7600119808
  memTotal: 8162086912
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.7.2-3.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.7.2
      commit: 0356bf4aff9a133d655dc13b1d9ac9424706cac4
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 2147483648
  swapTotal: 2147483648
  uptime: 6h 50m 42.00s (Approximately 0.25 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 600166400
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1000/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 4.3.1
  Built: 1668180253
  BuiltTime: Sat Nov 12 00:24:13 2022
  GitCommit: ""
  GoVersion: go1.18.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.3.1
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

_No response_

### Additional information

**Summary**: When running wsl --list --running, the running wsl distribution name is appended with **(default)**, causing  problems.  

There is a function to check whether wsl is normally running in **isWSLRunning** in **machine.go**

``` go
// machine.go
func isWSLRunning(dist string) (bool, error) {
       // Check the running wsl distribution.
	cmd := exec.Command("wsl", "-l", "--running") 
        ....

       // Receive cmd's stdout as a string and check if there is podman distribution text.
	scanner := bufio.NewScanner(transform.NewReader(out, unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder()))
	result := false
	for scanner.Scan() {
		fields := strings.Fields(scanner.Text())
		if len(fields) > 0 && dist == fields[0] {
			result = true
			break
		}
	}

}

```
Check the running wsl distro as a string value.  

However, if the actual podman is specified as the default value for wsl, the result will be false even though podman is operating normally.  

image

``` bat
> wsl --list --running
Linux용 Windows 하위 시스템 배포:
podman-machine-default(default)
```

Therefore, it would be nice to add the actual distribution name and the --quiet option to remove the header.

``` bat
> wsl --list --running --quiet
podman-machine-default
```

If there is no wsl distribution, the first wsl distribution is always set to the default.

### PR DESCRIPTION
**Pull request by shblue21**
If podman is set to the wsl default distribution, the (default) value is appended like podman-machine-default **(default)**.  

``` bat
> wsl --list --running
Linux용 Windows 하위 시스템 배포:
podman-machine-default(default)
```
Because of **(default)**, it determines that podman wsl deployment has not been executed.  
It can be solved simply by outputting only the wsl distribution name with the --quiet option.
``` bat
> wsl --list --running --quiet
podman-machine-default
```

Related to #17227, #17158  

Signed-off-by: shblue21 <jihunkimkw@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note
Fixed a startup issue with podman machine when running under some locales of Windows. 
```

[NO NEW TESTS NEEDED]

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Thanks for contributing!

In order to pass CI, could you add the magic string `[NO NEW TESTS NEEDED]` to your commit message and repush?

@n1hility PTAL

**Comment by n1hility** [comment]
Thanks for the PR! it looks like this is triggered when running under locales with different spacing rules (the previous code had a faulty assumption the wsl command output would always include a space between the name and the default signifier column). 

/approve

**Comment by n1hility** [comment]
I rebased the change with the no tests flag, so once this is green we can merge.

**Comment by shblue21** [comment]
@n1hility 
Ah, it must have been a locale-specific problem. I didn't know. Thank you
I use podman very usefully.
Thank you for your feedback.

**Comment by vrothberg** [comment]
> [+1563s] [Fail] Podman play kube [It] podman play kube test with hostIPC 
[+1563s] /var/tmp/go/src/github.com/containers/podman/test/e2e/play_kube_test.go:4992

It looks like a bad test was merged earlier.  I will look into it.

**Comment by vrothberg** [comment]
Please rebase once https://github.com/containers/podman/pull/17247 got merged and the test will pass again.  Sorry for the inconvenience.

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

**Comment by n1hility** [comment]
@shblue21 thanks again for the PR and using podman. We appreciate the help!

**Pull request merged**
